### ISSUE TITLE
Error after checking for tensorflow 2.10 installation

### ISSUE THREAD

**Issue by desmodus1984** [Gaius-Augustus/Tiberius#48]
Hi,
I tried running Tiberius and it failed several times. I tried installing tensorflow 2.10 as stated,
https://github.com/Gaius-Augustus/Tiberius/blob/main/docs/install_tensorflow.md

and after doing the installation check, I got the following error message in conda:

 python3 -c "import tensorflow as tf; print(tf.config.list_physical_devices('GPU'))"
2025-07-02 11:35:49.013384: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F AVX512_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2025-07-02 11:35:49.162298: I tensorflow/core/util/util.cc:169] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2025-07-02 11:35:49.192731: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2025-07-02 11:35:50.762821: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/data/common/juanpablo.aguilar/apps/miniconda3/lib/:/data/common/juanpablo.aguilar/apps/miniconda3/envs/tf2_10/lib/
2025-07-02 11:35:50.763065: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/data/common/juanpablo.aguilar/apps/miniconda3/lib/:/data/common/juanpablo.aguilar/apps/miniconda3/envs/tf2_10/lib/
2025-07-02 11:35:50.763100: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2025-07-02 11:35:51.768417: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/data/common/juanpablo.aguilar/apps/miniconda3/lib/:/data/common/juanpablo.aguilar/apps/miniconda3/envs/tf2_10/lib/
2025-07-02 11:35:51.768453: W tensorflow/stream_executor/cuda/cuda_driver.cc:263] failed call to cuInit: UNKNOWN ERROR (303)
2025-07-02 11:35:51.768489: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (pirate.cas.hamptonu.edu): /proc/driver/nvidia/version does not exist
[]

I did the installation for a 2nd time since I thought I did it wrong or there was something different from the other instruction - which are identical, (both links seem to be for conda and under a different python setup), and I got the error.

Did I miss something?

Hope this can be fixed.

**Comment by LarsGab** [comment]
Hi,

if I understand correctly, you haven't installed Tiberius itself yet and only installed Tensorflow 2.10 using the instructions?


Unfortunately, . I tested it on our machine and couldn’t reproduce the error, can you share your environment details (e.g., GPU model)?


If you have Singularity available, running the Singularity image is the easiest fix. Otherwise, upgrading to TensorFlow 2.17 and using a shorter sequence length (259,992) with tiberius.py should also work without any notable loss in accuracy. A newer Tensorflow version should be installed automatically, if you install Tiberius with `pip install .` .

Best,
Lars

**Comment by LarsGab** [comment]
Hi, 

the Singularity image has 11Gb. 

If you do not have a GPU, the installation of tf 2.10 should have worked and you can ignore the warnings. It just did not find a GPU. 

You can install Tiberius now with `pip install .` .

Best,
Lars
